### PR TITLE
Fixes CANCEL messages not working

### PR DESCRIPTION
**Pull request by kheimerl**
Tested as working with outbound. The CSeq change may not be necessary but is in protocol.

### CODE REVIEW

**Comment by kheimerl** [comment]
ptal

File: SIP/SIPMessage.cpp [hunk 1]
```diff
@@ -537,9 +537,14 @@ SipMessageRequestWithinDialog::SipMessageRequestWithinDialog(string reqMethod, S
...
 	this->msmFrom = *dialog->dsRequestFromHeader();
-	this->msmCSeqNum = dialog->dsNextCSeq();	// The BYE seq number must be incremental within the enclosing INVITE dialog.
+	if (incrementCSeq){ //NBYE and Dtmf increment, CANCEL does not -kurtis
+		this->msmCSeqNum = dialog->dsNextCSeq();	// The BYE seq number must be incremental within the enclosing INVITE dialog.
+	} else{
+		this->msmCSeqNum = dialog->mLocalCSeq; 
+	}
 	if (branch.empty()) { branch = make_branch(reqMethod.c_str()); }
-	this->smAddViaBranch(dialog,branch);
+	//this->smAddViaBranch(dialog,branch); // SHOULD NOT CREATE NEW VIA HEADER, USE OLD ONE -kurtis
```
Comment (shaddi) [inline SIP/SIPMessage.cpp]: nit: delete instead of cluttering with commented out code

**Comment by kheimerl** [comment]
cluttering with commented-out code is in the OpenBTS C++ best standard

**Comment by shaddi** [comment]
...I wasn't going to say it but that is true. Would prefer it were deleted still but otherwise lgtm.

**Pull request merged**